app/views/layouts/application.html.erb

```diff
@@ -21,16 +14,24 @@
     <link href='https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,700' rel='stylesheet' type='text/css'>
     <%= javascript_include_tag 'application' %>
     <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-Fy6S3B9q64WdZWQUiU+q4/2Lc9npb8tCaSX9FK7E8HnRr0Jz8D6OP9dO5Vg3Q9ct" crossorigin="anonymous"></script>
-    <% if Settings.GOOGLE_ANALYTICS_ID.present? %>
-    <script>
-    // Universal analytics
-    (function(i,s,o,g,r,a,m){i["GoogleAnalyticsObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,"script","//www.google-analytics.com/analytics.js","ga");
-
-    ga("create", "<%= Settings.GOOGLE_ANALYTICS_ID %>", "auto");
-    ga("send", "pageview");
-
+    
+    <!-- Google tag (gtag.js) --> 
+    <meta name="analytics_debug" value="<%= ENV.fetch('ANALYTICS_DEBUG', 1) %>">
+    <script async src="https://www.googletagmanager.com/gtag/js?id=G-4H0K34LTNE"></script> 
+    <script> 
+      const debug = document.head.querySelector("meta[name=analytics_debug]").getAttribute("value") === "1";
+      window.dataLayer = window.dataLayer || []; 
+      function gtag(){dataLayer.push(arguments);} 
+      // To turn off debug mode, exclude the parameter altogether (cannot just set to false)
+      // See https://support.google.com/analytics/answer/7201382?hl=en#zippy=%2Cgoogle-tag-websites
+      const args = {}
+      if (debug) {
```

Comment:
Instead of resolving this, can you instead have the ruby evaluate it directly? 
```
<% if ENV.fetch('ANALYTICS_DEBUG', '1') == '1' %>
args.debug_mode = true
<% end %>
```

Then you can also remove lines 19 & 22

Reply:
I changed the approach so hopefully this should be addressed as well.